addons/l10n_de/report/din5008_report.xml

```diff
@@ -24,73 +39,156 @@
-                        </t>
-                        <div class="company_address">
-                            <div>
-                                <strong t-field="company.partner_id.name"/>
-                            </div>
-
-                            <span t-field="company.partner_id"
-                                t-options='{"widget": "contact", "fields": ["address"], "no_marker": true}'/>
-                            <t t-if="information_block">
+                <div t-attf-class="din_page invoice_note article o_company_#{company.id}_layout" t-att-data-oe-model="o and o._name" t-att-data-oe-id="o and o.id">
+                    <table>
+                        <tr>
+                            <td>
                                 <div class="address">
-                                    <t t-raw="information_block"/>
+                                    <t t-if="company.name">
+                                        <span t-field="company.name"/>
+                                    </t>
+                                    <t t-if="company.street">
+                                        <span>|</span> <span t-field="company.street"/>
+                                    </t>
+                                    <t t-if="company.street2">
+                                        <span>|</span> <span t-field="company.street2"/>
+                                    </t>
+                                    <t t-if="company.zip">
+                                        <span>|</span> <span t-field="company.zip"/>
+                                    </t>
+                                    <t t-if="company.city">
+                                        <span t-if="not company.zip">|</span> <span t-field="company.city"/>
+                                    </t>
+                                    <t t-if="company.country_id">
+                                        <span>|</span> <span t-field="company.country_id.name"/>
+                                    </t>
+                                    <hr class="company_invoice_line" />
+                                    <div t-if="address">
+                                        <t t-raw="address"/>
+                                    </div>
+                                    <div t-else="fallback_address">
+                                        <t t-raw="fallback_address"
+                                           t-options='{"widget": "contact", "fields": ["address", "name", "phone"], "no_marker": True}' />
+                                    </div>
                                 </div>
-                            </t>
-                        </div>
-                    </div>
-                        <t t-raw="0" />
+                            </td>
+                            <td>
+                                <div class="information_block" t-if="o">
+                                    <table t-if="o._name == 'account.move'">
```

Comment:
o_O

we need to discuss that

Reply:
Templates are sh*t, but there is nothing we can change about that in stable...
Or I could add new modules for stock, sale, purchase, etc? There would still be this weird that Ng but at least with the models at the "right place".
Then there is also the base template model that is weird...

The main issue is that I have to display fields that are part from the document (invoice, SO, PO, ...) in the template directly.